crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -374,6 +381,117 @@ impl Persistence {
 
         Ok(solution)
     }
+
+    /// Returns the oldest settlement event for which the accociated auction is
+    /// not yet populated in the database.
+    pub async fn get_settlement_without_auction(
+        &self,
+    ) -> Result<Option<domain::eth::Event>, DatabaseError> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["get_settlement_without_auction"])
+            .start_timer();
+
+        let mut ex = self.postgres.pool.begin().await?;
```

Comment:
We don't need to use a transaction here, right?
```suggestion
        let mut ex = self.postgres.pool.acquire().await?;
```